ai-intro/ai-intro-quiz-notes.md

```diff
@@ -0,0 +1,57 @@
+# [template]-quiz-notes
+
+## Quiz Questions
+
+Answer the following questions in the provided markdown file before turning in this exercise:
+
+- Will AI replace software developers?
+
+No, because AI has learned what it knows from programmers & documentation, it essentially makes predictions. AI can predict based on patterns and trends it has seen a thousand times over but may not be able to understand how multiple things can be implemented together. It is useful to write smaller blocks rudimentary of code.
+
+AI is unlikely to fully replace software developers in the foreseeable future. While AI can automate certain coding tasks, optimize workflows, and even generate code snippets, software development requires creativity, critical thinking, and an understanding of complex systems, which AI lacks. Developers also engage in tasks like system design, user experience considerations, and problem-solving in ways that are beyond AI’s current capabilities. Instead, AI is expected to serve as a powerful tool that enhances developers' productivity rather than replacing them altogether.
+
+My observation: my answer and AI's share similarities but the AI's is more fleshed out and wordy. I have noticed with AI, especially ChatGPT the answers are not succinct and that is how I usually detect that AI was utilized.
+
+- What is the best way to use AI tools when learning software development?
+
+You can use AI best by asking questions about the functionality, syntax, concepts, etc of code, but not to write it as it's still relatively young in its coding expertise.
+
+Learning assistance, debugging, code exploration, project prototyping
+
+My observation: AI provided several main themes to the question proposed and explained each thoroughly, mentioning more best practices than my initial answer did (I only pasted the topics).
+
+- What is the danger of using AI tools for software development while I'm still learning?
+
+The danger is that you will not learn the fundamentals on your own thereby, if you are unable to read and detect the errors in code and AI’s regurgitation, then you most likely should not be utilizing AI into your coding efforts.
+
+Over-reliance, bad practices, reduced problem-solving skills, security and ethical issues
+
+My observation: AI mentioned the matter of security and ethics in its response which I did not read in the instructions, which is a valid matter to bring attention to.
```

Comment:
Very valid! If you paste in snippets from your companies code base into chatGPT, now they are training their language models off of your companies code!